### PR TITLE
test_primary_storage_scope.py should only run with kvm, vmware and simulator

### PR DESCRIPTION
**Pull request by abh1sar**
### Description

This PR fixes the smoke failure in test_primary_storage_scope.py when run with XenServer.
#9315 

```
[SF] Trillian test result (tid-10688)
Environment: xenserver-71 (x2), Advanced Networking with Mgmt server 7
Total time taken: 41254 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9315-t10688-xenserver-71.zip
Smoke tests completed. 131 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:

Test	Result	Time (s)	Test File
test_01_primary_storage_scope_change	Error	0.07	test_primary_storage_scope.py
```
Zone wide storage pools are not supported with Xenserver so storage pool creation was failing during setup, and not cleaning up the created cluster.
This causes cluster create to fail if the test is retried.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [x] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by abh1sar** [comment]
@blueorangutan package

**Comment by abh1sar** [comment]
@blueorangutan package

**Comment by abh1sar** [comment]
@blueorangutan test matrix

**Comment by sureshanaparti** [comment]
@blueorangutan test matrix

File: test/integration/smoke/test_primary_storage_scope.py [hunk 1]
```diff
@@ -35,22 +35,41 @@ def setUp(self):
...
             'clustertype': 'CloudManaged'
         }
+        return
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
```
[the remark this answers is not in the document]
Comment (abh1sar) [reply test/integration/smoke/test_primary_storage_scope.py]: Done

**Comment by abh1sar** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test matrix

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**